src/software/geom/util.cpp

```diff
@@ -1350,75 +1348,88 @@ std::vector<Circle> findOpenCircles(Rectangle rectangle, std::vector<Point> poin
     // on the field, and the size of the circle is the distance to the closest vertex
     // on the triangle that this vertex was created from
 
-    // TODO: instead of just adding the 4 corners of the rectangle here, the proper
-    //       thing to do is to generate the voronoi with just the points given, then
-    //       also treat all intersections of the voronoi diagram with the edge of the
-    //       rectangle as vertices
-    //       https://stackoverflow.com/questions/34093602/efficient-algorithm-to-determine-largest-open-space
+    // Filters out points that are outside of the bounding box
+    points.erase(std::remove_if(points.begin(), points.end(),
+                                [&bounding_box](const Point &p) {
+                                    return !bounding_box.containsPoint(p);
+                                }),
+                 points.end());
 
-    // Add the 4 corners of the rectangle to the list of points
-    points.emplace_back(rectangle.posXPosYCorner());
-    points.emplace_back(rectangle.negXPosYCorner());
-    points.emplace_back(rectangle.posXNegYCorner());
-    points.emplace_back(rectangle.negXNegYCorner());
+    std::vector<Circle> empty_circles;
+
+    // Creating the Voronoi diagram with 2 or less points will produce no edges so we need
+    // to handle these cases manually
+    if (points.empty())
+    {
+        // If there are no points, return an empty vector since there are no constraints
+        // to the size of the circle.
+        return empty_circles;
+    }
+    else if (points.size() == 1)
```

Comment:
nit: we don't need `else if`, just an `if` would suffice

Reply:
done